controllers/vcdmachine_controller.go

```diff
@@ -469,9 +470,16 @@ func (r *VCDMachineReconciler) reconcileVMBootstrap(ctx context.Context, vcdClie
 
 				return errors.Wrapf(err, "Error while generating network initialization script for ignition [%s/%s]", vcdCluster.Name, vm.VM.Name)
 			}
+
+			// GZip then Base64 encode bootstrap data
+			gzipb64encodedData, err := gzipAndBase64Encode([]byte(bootstrapData))
+			if err != nil {
+				log.Error(err, "Failed to gzip and base64 encode data")
```

Comment:
In case there is any other data that could be helpful in this error message, I'd suggest using `return errors.Wrapf()` here and including it

Reply:
Would you mind showing me an example to do it here?

Reply:
Ah yeah for example in the previous line 472 we include information on the ignition like this
```
return errors.Wrapf(err, "Error while generating network initialization script for ignition [%s/%s]", vcdCluster.Name, vm.VM.Name)
```
The function wraps the original error and adds the context (...for ignition vcdclustername/vmname) at the end.
Because I'm not so familiar with this controller I don't know if any extra content would be helpful for debugging. However if it is, it could look something like this if we want to refer to the ignition in the same way.

```
return errors.Wrapf(err, "Failed to gzip and base64 encode bootstrap data for ignition [%s/%s]", vcdCluster.Name, vm.VM.Name)
```
Or we could add anything else other than `vcdCluster.Name` and `vm.VM.Name` as well if it makes sense.

Reply:
Nice, addressed in https://github.com/giantswarm/cluster-api-provider-cloud-director/pull/18/commits/457e5cd731c8938877e43432f7fcb6838d223d81